data-prepper-plugins/aggregate-processor/README.md

```diff
@@ -1,29 +1,153 @@
+
+## Available Aggregate Actions
+
+### <a name="remove_duplicates"></a>
+* `remove_duplicates`: Processes the first Event for a group immediately, and drops Events that follow for that group.
+  * After the following Event is processed with `identification_keys: ["sourceIp", "destination_ip"]`:
+      ```json
+          { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "status": 200 }
+      ```
+    The following Event will be dropped by Data Prepper:
+      ```json
+         { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 1000 }
+      ```
+    While this Event will be processed by Data Prepper and create a new group (since the `sourceIp` is different):
+      ```json
+         { "sourceIp": "127.0.0.2", "destinationIp": "192.168.0.1", "bytes": 1000 }
+      ```
+
+### <a name="put_all"></a>
+* `put_all`: Combine Events belonging to the same group by overwriting existing keys and adding non-existing keys (the equivalence of java `Map.putAll`). All Events that make up the combined Event will be dropped.
+    * Given the following three Events with `identification_keys: ["sourceIp", "destination_ip"]`:
+      ```json
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "status": 200 }
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 1000 }
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "http_verb": "GET" }
+      ```
+      The following Event will be created and processed by the rest of the pipeline when the group is concluded:
+      ```json
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "status": 200, "bytes": 1000, "http_verb": "GET" }
+      ```
+
+## Creating New Aggregate Actions
+
+It is easy to create custom Aggregate Actions to be used by the Aggregate Processor. To do so, create a new class that implements the [AggregateAction interface](src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateAction.java).
+The interface has the following signature:
+
+```
+public interface AggregateAction {
+
+    // This function will be called once for every single Event. If this function throws an error, the Event passed will be returned and processed by the rest of the pipleine.
+    default AggregateActionResponse handleEvent(final Event event, final AggregateActionInput aggregateActionInput) {
+        return AggregateActionResponse.fromEvent(event);
+    }
+    
+    // This function will be called once for a group. It will be called when the group_duration for that group has passed.
+    default Optional<Event> concludeGroup(final AggregateActionInput aggregateActionInput) {
+        return Optional.empty();
+    }
+}
+```
+
+The `AggregateActionInput` that is passed to the functions of the interface contains a method `getGroupState()`, which returns a `GroupState` Object that can be operated on like a java `Map`. 
+Here is an example of an `AggregateAction` implementation for [put_all](#put_all):
```

Comment:
Should we just link them to the current file instead? These files seem even more likely to change than the `AggregateAction` interface.

It might be nice to link readers to the directory which has all the current actions. And let them know that they can look at the different implementations in there.